test/functional/fixtures/page-js-errors/test.js

```diff
@@ -56,11 +66,226 @@ describe('Should ignore an js-error on the page if the skipJsErrors option is se
+            return runTests('./testcafe-fixtures/test-controller.js', 'Should skip JS errors with multiple options');
+        });
+
+        it('Should skip JS errors with callback function', async () => {
+            return runTests('./testcafe-fixtures/test-controller.js', 'Should skip JS errors with callback function');
+        });
+
+        it('Should skip JS errors with regular expression in message', async () => {
+            return runTests('./testcafe-fixtures/test-controller.js', 'Should skip JS errors with regular expression in message');
+        });
+
+        it('Should fail if the regexp message option doesnt satisfy the client error message', async () => {
+            return runTests('./testcafe-fixtures/test-controller.js', 'Should fail if the regexp message option doesnt satisfy the client error message', { shouldFail: true })
+                .catch(errs => {
+                    expectFailAttempt(errs, CLIENT_ERROR_MESSAGE);
+                });
+        });
+
+        it('Should fail if the message option doesnt satisfy the client error message', async () => {
+            return runTests('./testcafe-fixtures/test-controller.js', 'Should fail due to incorrect message option', { shouldFail: true })
+                .catch(errs => {
+                    expectFailAttempt(errs, CLIENT_ERROR_MESSAGE);
+                });
+        });
+
+        it('Should fail if callback function logic doesnt satisfy the client error message', async () => {
+            return runTests('./testcafe-fixtures/test-controller.js', 'Should fail with callback function', { shouldFail: true })
+                .catch(errs => {
+                    expectFailAttempt(errs, CLIENT_ERROR_MESSAGE);
+                });
+        });
+
+        it('Should fail if error occurred in callback function', async () => {
+            return runTests('./testcafe-fixtures/test-controller.js', 'Should fail due to error in callback function', { shouldFail: true })
+                .catch(errs => {
+                    expectFailAttempt(errs, CALLBACK_FUNC_ERROR);
+                });
+        });
+
+        it('Should skip JS errors with async callback function', async () => {
+            return runTests('./testcafe-fixtures/test-controller.js', 'Should skip JS errors, with async callback function', { skip: ['ie'] });
+        });
+
+        it('Should fail if async callback function does not satisfy the error message', async () => {
+            return runTests('./testcafe-fixtures/test-controller.js', 'Should fail if async callback function does not satisfy the error message', {
+                shouldFail: true,
+                skip:       ['ie'],
+            })
+                .catch(errs => {
+                    expectFailAttempt(errs, CLIENT_ERROR_MESSAGE);
+                });
+        });
+    });
+
+    describe('Fixture and Test method', () => {
+        it('Should skip JS errors with callback function specified in fixture', () => {
```

Comment:
The core logic was checked in the `TestController` section.
You can leave only 2 tests: `test.skipJsErrors` and `fixture.skipJsErrors`.